### PR TITLE
update to latest jda version

### PR DESCRIPTION
**Pull request by 0ptk**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing
[pull-request]: https://github.com/JDA-Applications/JDA-Utilities/pulls

#### Pull Request Checklist
Please follow the following steps before opening this PR.<br>
PRs that do not complete the checklist will be subject to denial for
missing information.

- [X] I have checked the [pull request page][pull-request] for upcoming
      or merged features/bug fixes.
- [X] I have read JDA's [contributing guidelines][contributing].

#### Pull Request Information
Check and fill in the blanks for all that apply:

- [X] My PR fixes a bug, error, or other issue with the library's codebase.
- [ ] My PR is for the `______` module of the JDA-Utilities library.
- [ ] My PR creates a new module for the JDA-Utilities library: `______`.

#### Description

I updated jda version to latest and small fix codes

### CODE REVIEW

**Comment by astralship-system** [comment]
JDA-Utilities has been dead since quite a while, so I doubt this will be merged anytime soon. You might want to look into forks or even a different library.

**Comment by Sanduhr32** [comment]
You broke several core functionalities of "Builder" classes by returning void disallowing chaining. This PR will most likely not be merged unless you undo most changes breaking how JDA Utils is set up.

**Comment by Andre601** [comment]
As it was mentioned is this project abandoned.
You should consider using [`chew/JDA-Chewtils`](https://github.com/Chew/JDA-Chewtils) which is updated and maintsined.

Also, as sanduhr mentioned does your PR also break everything.

**Comment by Sanduhr32** [comment]
English? Do you speak it? @Andre601 Andtre you must formulate declarative sentences not questions.

**Comment by Andre601** [comment]
> English? Do you speak it? @Andre601 Andtre you must formulate declarative sentences not questions.

Shush. I don't complain about your speech-skills so I expect the same from you.
My points where clear enough.

**Comment by Sanduhr32** [comment]
This was a friendly reminder that `verb noun` most likely is used for questioning. You do not try to question stuff right now your answer is a redundancy with incorrect language.
I hereby politely ask you to fix your incorrect chaining of verb and noun to noun and verb.

File: command/src/main/java/com/jagrosh/jdautilities/command/Command.java [hunk 1]
```diff
@@ -557,10 +557,12 @@ public String getCooldownKey(CommandEvent event)
             case GUILD:        return event.getGuild()!=null ? cooldownScope.genKey(name,event.getGuild().getIdLong()) :
                     CooldownScope.CHANNEL.genKey(name,event.getChannel().getIdLong());
             case CHANNEL:      return cooldownScope.genKey(name,event.getChannel().getIdLong());
-            case SHARD:        return event.getJDA().getShardInfo()!=null ? cooldownScope.genKey(name, event.getJDA().getShardInfo().getShardId()) :
-                    CooldownScope.GLOBAL.genKey(name, 0);
-            case USER_SHARD:   return event.getJDA().getShardInfo()!=null ? cooldownScope.genKey(name,event.getAuthor().getIdLong(),event.getJDA().getShardInfo().getShardId()) :
-                    CooldownScope.USER.genKey(name, event.getAuthor().getIdLong());
+            case SHARD:
+                event.getJDA().getShardInfo();
+                return cooldownScope.genKey(name, event.getJDA().getShardInfo().getShardId());
```
Comment (MinnDevelopment) [inline command/src/main/java/com/jagrosh/jdautilities/command/Command.java:562]: This change makes no sense and breaks logic

File: command/src/main/java/com/jagrosh/jdautilities/command/Command.java [hunk 2]
```diff
@@ -557,10 +557,12 @@ public String getCooldownKey(CommandEvent event)
...
+            case USER_SHARD:
+                event.getJDA().getShardInfo();
+                return cooldownScope.genKey(name,event.getAuthor().getIdLong(),event.getJDA().getShardInfo().getShardId());
```
Comment (MinnDevelopment) [inline command/src/main/java/com/jagrosh/jdautilities/command/Command.java:565]: This change makes no sense and breaks logic

File: command/src/main/java/com/jagrosh/jdautilities/command/CommandBuilder.java [hunk 3]
```diff
@@ -64,15 +64,13 @@
      * @param  name
      *         The name of the Command to be built.
      *
-     * @return This CommandBuilder
      */
-    public CommandBuilder setName(String name)
+    public void setName(String name)
```
Comment (MinnDevelopment) [inline command/src/main/java/com/jagrosh/jdautilities/command/CommandBuilder.java:68]: This is a breaking change and makes the builder less useful as you remove the fluent interface pattern for no reason.

File: command/src/main/java/com/jagrosh/jdautilities/command/CommandEvent.java [hunk 4]
```diff
@@ -1031,7 +1031,8 @@ public SelfUser getSelfUser()
      */
     public Member getSelfMember()
     {
-        return event.getGuild() == null ? null : event.getGuild().getSelfMember();
+        event.getGuild();
+        return event.getGuild().getSelfMember();
```
Comment (MinnDevelopment) [inline command/src/main/java/com/jagrosh/jdautilities/command/CommandEvent.java:1035]: Again this change makes no sense and breaks logic

File: command/src/main/java/com/jagrosh/jdautilities/command/impl/CommandClientImpl.java [hunk 5]
```diff
@@ -573,12 +574,10 @@ private void sendStats(JDA jda)
...
-            
-            if(jda.getShardInfo() != null)
-            {
-                bodyBuilder.add("shard_id", Integer.toString(jda.getShardInfo().getShardId()))
-                           .add("shard_count", Integer.toString(jda.getShardInfo().getShardTotal()));
-            }
+
+            jda.getShardInfo();
+            bodyBuilder.add("shard_id", Integer.toString(jda.getShardInfo().getShardId()))
+                       .add("shard_count", Integer.toString(jda.getShardInfo().getShardTotal()));
```
Comment (MinnDevelopment) [inline command/src/main/java/com/jagrosh/jdautilities/command/impl/CommandClientImpl.java:580]: Again, nonsense change.

File: command/src/main/java/com/jagrosh/jdautilities/command/impl/CommandClientImpl.java [hunk 6]
```diff
@@ -587,14 +586,14 @@ private void sendStats(JDA jda)
...
-                public void onResponse(Call call, Response response)
+                public void onResponse(@NotNull Call call, @NotNull Response response)
                 {
                     LOG.info("Successfully send information to carbonitex.net");
                     response.close();
                 }
 
                 @Override
-                public void onFailure(Call call, IOException e)
+                public void onFailure(@NotNull Call call, @NotNull IOException e)
```
Comment (MinnDevelopment) [inline command/src/main/java/com/jagrosh/jdautilities/command/impl/CommandClientImpl.java:596]: These annotations do nothing.

File: command/src/main/java/com/jagrosh/jdautilities/command/impl/CommandClientImpl.java [hunk 7]
```diff
@@ -619,12 +616,12 @@ public void onFailure(Call call, IOException e)
...
             {
                 @Override
-                public void onResponse(Call call, Response response) throws IOException
+                public void onResponse(@NotNull Call call, @NotNull Response response) throws IOException
                 {
                     if(response.isSuccessful())
                     {
                         LOG.info("Successfully sent information to discord.bots.gg");
-                        try(Reader reader = response.body().charStream())
+                        try(Reader reader = Objects.requireNonNull(response.body()).charStream())
```
Comment (MinnDevelopment) [inline command/src/main/java/com/jagrosh/jdautilities/command/impl/CommandClientImpl.java:624]: requreNonNull is useless here

File: commons/src/main/java/com/jagrosh/jdautilities/commons/JDAUtilitiesInfo.java [hunk 8]
```diff
@@ -28,13 +28,12 @@
     public static final String VERSION_MAJOR = "@VERSION_MAJOR@";
     public static final String VERSION_MINOR = "@VERSION_MINOR@";
     public static final String VERSION_REVISION  = "@VERSION_REVISION@";
-    public static final String VERSION = VERSION_MAJOR.startsWith("@")? "DEV" : VERSION_MAJOR + "." + VERSION_MINOR + "." + VERSION_REVISION;
+    public static final String VERSION = "DEV";
```
Comment (MinnDevelopment) [inline commons/src/main/java/com/jagrosh/jdautilities/commons/JDAUtilitiesInfo.java:31]: This change breaks versions entirely

File: menu/src/main/java/com/jagrosh/jdautilities/menu/ButtonMenu.java [hunk 9]
```diff
@@ -112,36 +113,35 @@ private void initialize(RestAction<Message> ra)
                 else
                 {
                     // This is the last reaction added.
-                    r.queue(v -> {
-                        waiter.waitForEvent(MessageReactionAddEvent.class, event -> {
-                            // If the message is not the same as the ButtonMenu
-                            // currently being displayed.
-                            if(!event.getMessageId().equals(m.getId()))
-                                return false;
+                    r.queue(v -> waiter.waitForEvent(MessageReactionAddEvent.class, event -> {
```
Comment (MinnDevelopment) [inline menu/src/main/java/com/jagrosh/jdautilities/menu/ButtonMenu.java:116]: This change makes the code harder to read

File: menu/src/main/java/com/jagrosh/jdautilities/menu/Menu.java [hunk 10]
```diff
@@ -153,7 +152,7 @@ protected boolean isValidUser(User user, @Nullable Guild guild)
         if(guild == null || !guild.isMember(user))
             return false;
 
-        return guild.getMember(user).getRoles().stream().anyMatch(roles::contains);
+        return Objects.requireNonNull(guild.getMember(user)).getRoles().stream().anyMatch(roles::contains);
```
Comment (MinnDevelopment) [inline menu/src/main/java/com/jagrosh/jdautilities/menu/Menu.java:155]: Useless

File: oauth2/src/main/java/com/jagrosh/jdautilities/oauth2/requests/OAuth2Requester.java [hunk 11]
```diff
@@ -44,44 +45,31 @@ public OAuth2Requester(OkHttpClient httpClient)
...
-                    }
-                    catch(Throwable t)
-                    {
+                    } catch (Throwable t) {
                         LOGGER.error("OAuth2Action success callback threw an exception!", t);
                     }
-                }
-                catch(Throwable t)
-                {
+                } catch (Throwable t) {
```
Comment (MinnDevelopment) [inline oauth2/src/main/java/com/jagrosh/jdautilities/oauth2/requests/OAuth2Requester.java:61]: You are changing code style for no reason

**Comment by MinnDevelopment** [comment]
Closing since the majority of changes introduced by this PR are completely unrelated to updating JDA. They are mostly superfluous and in some cases outright breaking and even introduce bugs.

**Pull request closed without merging**